File: Examples/CSharp/Articles/RenderProjectToPredefinedPageSizes.cs
/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from https://www.nuget.org/packages/Aspose.Tasks/, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using https://forum.aspose.com/c/tasks
*/

namespace Aspose.Tasks.Examples.CSharp.Articles
{
    using System;

    using Aspose.Tasks.Saving;
    using Aspose.Tasks.Visualization;

    internal class RenderProjectToPredefinedPageSizes
    {
        public static void Run()
        {  
            // The path to the documents directory.
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);

            //ExStart:RenderProjectToPredefinedPageSizes
            //ExFor: SaveOptions.PageSize
            //ExSummary: Shows how to set page size (can be one of the values of the <see cref="P:Aspose.Tasks.Visualization.TiffCompression" /> enumeration).
            var project = new Project(dataDir + "Project2.mpp");

            const PresentationFormat Format = PresentationFormat.GanttChart;

            // Render the project to all Pre-Defined page sizes
            foreach (var pageSize in (PageSize[])Enum.GetValues(typeof(PageSize)))
            {
                var options = new PdfSaveOptions();
                options.PresentationFormat = Format;
                options.FitContent = true;
                options.PageSize = pageSize;
                var resultFile = "result_PredefinedPageSizes_" + Format + "_" + pageSize + "_out.pdf";
                project.Save(dataDir + resultFile, options);
            }
            //ExEnd:RenderProjectToPredefinedPageSizes
        }
    }
}

File: Examples/VisualBasic/Articles/RenderProjectToPredefinedPageSizes.vb
Imports Aspose.Tasks.Visualization
Imports Aspose.Tasks.Saving

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Articles
    Public Class RenderProjectToPredefinedPageSizes
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:RenderProjectToPredefinedPageSizes
            ' Source File to Render
            Dim fileName As String = "Project2.mpp"

            ' Read the Project
            Dim project1 As New Project(dataDir & fileName)

            ' Select a Presentation format
            Dim format As PresentationFormat = PresentationFormat.GanttChart

            ' Render the project to all Pre-Defined page sizes
            For Each pageSize As PageSize In DirectCast([Enum].GetValues(GetType(PageSize)), PageSize())
                Dim options As New PdfSaveOptions()
                options.PresentationFormat = format
                options.FitContent = True
                options.PageSize = pageSize
                Dim resultFile As String = "result_PredefinedPageSizes_" & format.ToString() & "_" & pageSize.ToString() & "_out.pdf"
                project1.Save(dataDir & resultFile, options)
            Next
            ' ExEnd:RenderProjectToPredefinedPageSizes
        End Sub
    End Class
End Namespace

